tasks/gpsr/find_object/src/object_comparison.py

```diff
@@ -0,0 +1,175 @@
+            print("we're doing weight")
+            weights_dict = self.get_weight(userdata.detections_types)
+            sorted_weights = sorted(weights_dict.items(), key=lambda item: item[1], reverse= True)
+            userdata.sorted_weights = sorted_weights
+            print(sorted_weights)
+            return 'succeeded'
+
+    class ObjectSize(smach.State):
+        def __init__(self):
+            smach.State.__init__(self, outcomes=['succeeded', 'failed'], input_keys=['object_dict', 'detections_3d'], output_keys=['sorted_size'])
+
+        def property_size_calculation(self, detections, result):
+            """
+            calculate the size of a list of objects using bounding box.
+            :param detections: List of detection tuples (object_type, position)
+            :return: a list of the size of each object in a category 
+            """
+        # userdata.detections_3d = [(detection, point),(detection, point),(detection, point)]
+        # the detection includes the size of the bounding box and the center point of the box
+            area = dict()
+            for i in detections:
+                for object in result.detected_objects:
+                    if i == object.name:
+                        area[i] = object.xywh[2]*object.xywh[3]
+            return area  
+        
+        def execute(self, userdata):
+            # got the area dic of the detected objects
+            # rank the size of each object
+            print("we're doing size")
+            detections_types = userdata.object_dict.keys()
+            area_dict = self.property_size_calculation(detections_types, userdata.detections_3d)
+            sorted_size = sorted(area_dict.items(), key=lambda item: item[1], reverse = True)
+            userdata.sorted_size = sorted_size
+            print(sorted_size)
+            return 'succeeded'
+        
+    def __init__(
+        self,
+        area_polygon: Polygon,
+        depth_topic: str = "/xtion/depth_registered/points",
+        model: str = "yolov8n-seg.pt",
+        filter: Union[List[str], None] = None,
+        confidence: float = 0.5,
+        nms: float = 0.3
+    ):
+        smach.StateMachine.__init__(
+            self,
+            outcomes=["succeeded", "failed"],
+            output_keys=["detections_3d"],
+        )
+
+        with self:
+            smach.StateMachine.add(
+                "DETECT_OBJECTS_3D",
+                Detect3D(
+                    depth_topic=depth_topic,
+                    model=model,
+                    filter=filter,
+                    confidence=confidence,
+                    nms=nms,
+                ),
+                transitions={"succeeded": "DETECTIONS3D", "failed": "failed"},
+            )
+            # smach.StateMachine.add(
+            #     "FILTER_DETECTIONS",
+            #     Detect3DInArea(area_polygon),
+            #     transitions={"succeeded": 'DETECTIONS3D', "failed": "failed"},
+            # )
```

Comment:
```suggestion
                Detect3DInArea(
                    area_polygon
                    depth_topic=depth_topic,
                    model=model,
                    filter=filter,
                    confidence=confidence,
                    nms=nms,
                ),
                transitions={"succeeded": "DETECTIONS3D", "failed": "failed"},
            )
```

I think we should use the `Detect3DInArea` skill here now!